torch/csrc/jit/tensorexpr/eval.cpp

```diff
@@ -1278,13 +1278,24 @@ void SimpleIREvaluator::bindArg(const BufferArg& bufArg, void* data) {
   }
 
   switch (bufArg.dtype().scalar_type()) {
-#define TYPE_CASE(Type, Name)                 \
-  case ScalarType::Name: {                    \
-    Type typed_data;                          \
-    memcpy(&typed_data, data, sizeof(Type));  \
-    impl_->bindVar(bufArg.var(), typed_data); \
-    break;                                    \
-  }
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define TYPE_CASE(Type, Name)                                                          \
+  case ScalarType::Name: {                                                             \
+    Type typed_data;                                                                   \
+    memcpy(&typed_data, data, sizeof(Type));                                           \
+    impl_->bindVar(bufArg.var(), typed_data);                                          \
+    break;                                                                             \
+  }
+#else /* __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__ */
+#define TYPE_CASE(Type, Name)                                                          \
+  case ScalarType::Name: {                                                             \
+    Type typed_data;                                                                   \
+    size_t idx = (sizeof(Type) >= sizeof(void*)) ? 0 : (sizeof(void*) - sizeof(Type)); \
```

Comment:
I suppose little-endian and big-endian have the same bit-width, and the only difference is how to store the data. Regarding this change, it seems like it serves not only big-endian but also a new platform. Right? Because I do not quite understand why you need to compare with `sizeof(void*)`.

Reply:
This change is intended for existing big endian platforms and tested only on s390x.
I guess it might be possible to simplify this expression if we assume that sizeof(Type) always <= sizeof(void*).
I'll update this change.

Reply:
@AlekseiNikiforovIBM , the `sizeof` a pointer could be 32bit on a 32bit system while its bit width should be less than `sizeof(int64_t)`.